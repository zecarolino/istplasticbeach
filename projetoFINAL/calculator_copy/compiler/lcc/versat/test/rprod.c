/* multiply two numbers using a recursive function */
int product(int n1, int n2)
{
	if (n2 > 0) return (n1 + product(n1, n2 - 1));
	if (n2 < 0) return -product(n1, -n2);
	return 0;
}

int main() { return product(3, 4); }
