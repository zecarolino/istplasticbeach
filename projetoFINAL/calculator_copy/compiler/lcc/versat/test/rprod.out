VCD info: dumpfile xtop.vcd opened for output.
<
Catched Trap at time  1005 clock cycles:
